l3wrapper/l3wrapper.py

```diff
@@ -294,6 +300,9 @@ def fit(self, X, y, column_names=None, save_human_readable=False, remove_files=T
         if remove_files:
             _remove_fit_files(token)
 
+        chdir(old_dir)
+        tmp_dir.cleanup()
+
```

Comment:
I can still see some problems here. If a client chooses to **save_human_readable** format of the rules, the cleaning process of the temporary directory should not happen, otherwise, we lose them.

Maybe something like that?

```suggestion
        if remove_files:
            _remove_fit_files(token)

        chdir(old_dir)
        if remove_files and not save_human_readable:
            tmp_dir.cleanup()
```